Thimble issue1763 - Adding whitespace indicators to the editor

tonypark0403 opened the pull request:
Added whitespace by default

tonypark0403 commented:
This issue comes from https://github.com/mozilla/thimble.mozilla.org/issues/1763.
And I added the whitespace indicator by default.

humphd commented on Gruntfile.js:
```diff
@@ -522,4 +523,4 @@ module.exports = function (grunt) {
     // Default task.
     grunt.registerTask('default', ['test']);
-};
+};
```
Here's an example of what we discussed in class about not having an end-of-file line break.  Please revert this change back.

humphd commented on package.json:
```diff
@@ -70,4 +70,4 @@
     },
     "license": "MIT"
-}
+}
```
Same issue here, please revert.

humphd commented on src/config.json:
```diff
@@ -37,59 +37,50 @@
-        "tar-fs": "^1.11.1"
+        "requirejs": "^2.3.3",
+        "sw-precache": "^5.0.0",
+        "tar-fs": "^1.15.1"
```
All of these changes need to be reverted--it looks like you need to rebase.

humphd commented on src/utils/BrambleExtensionLoader.js:
```diff
@@ -34,6 +34,7 @@ define(function (require, exports, module) {
         "bramble",
         "Autosave",
         "brackets-paste-and-indent",
+        "brackets-show-whitespace",
```
I have serious misgivings about making this a default that we load.  @flukeout this needs to be tested by you, and I also want to do performance testing to make sure it doesn't bog down the editor.

humphd commented:
@flukeout can you test/review this?  Especially pay attention to perf.

flukeout commented:
Will do!

flukeout commented:
@tonypark0403 I'm not able to see the white space indicators in the Editor. I pulled your changes into brackets and then ran ``npm run build``. In the URL i also added ``?enableExtensions=brackets-show-whitespace`` - is there a step I am missing to see these changes?

cc @humphd - did you get it running?

tonypark0403 commented:
Yes, it is.
Since it is not decided to provide the feature by default, I deleted the line.
You can refer to this address, https://github.com/mozilla/brackets/pull/630/files/ddbe293cb1c1e36f04535fad130d4e55d8510098

In the file, src/utils/BrambleExtensionLoader.js, you should add "brackets-show-whitespace" in the 36 line.

From: Luke Pacholski [mailto:notifications@github.com]
Sent: Wednesday, March 08, 2017 2:59 PM
To: mozilla/brackets <brackets@noreply.github.com>
Cc: Heetai Park <hpark77@myseneca.ca>; Mention <mention@noreply.github.com>
Subject: Re: [mozilla/brackets] Thimble issue1763 - Adding whitespace indicators to the editor (#630)


@tonypark0403<https://github.com/tonypark0403> I'm not able to see the white space indicators in the Editor. I pulled your changes into brackets and then ran npm run build. In the URL i also added ?enableExtensions=brackets-show-whitespace - is there a step I am missing to see these changes?

cc @humphd<https://github.com/humphd> - did you get it running?

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/mozilla/brackets/pull/630#issuecomment-285151286>, or mute the thread<https://github.com/notifications/unsubscribe-auth/ANrzz9ryiFqVGn9OQ2ZoQ8avkWvzHBXiks5rjwh3gaJpZM4MVAPM>.

flukeout commented:
Thanks @tonypark0403 - trying it out now.

flukeout commented:
Hi @tonypark0403 - I added the extension in ``src/utils/BrambleExtensionLoader.js``...

![image](https://cloud.githubusercontent.com/assets/25212/23776625/3a5b2302-04e4-11e7-87e7-725a6b5a1446.png)

...but still don't see any whitespaces.

![image](https://cloud.githubusercontent.com/assets/25212/23776613/2230fd2e-04e4-11e7-86d5-25b25975caaf.png)

tonypark0403 commented:
I added the tmpl files and applied it. Try it.

flukeout commented:
Will try now - thank you.

flukeout commented:
This works great @tonypark0403 - nice work! Finally got everything running. Ultimately, I think this should be an option in the gear menu dropdown and should be turned on by default, like this...

![image](https://cloud.githubusercontent.com/assets/25212/23868139/936b9c48-07db-11e7-8728-2de787da16a2.png)

However, to get there I think we should...

* First, land this pull request but disable the extension by default
* Second, start a new issue related to adding it to the menu

@tonypark0403 - can you change it so that the extension is NOT enabled by default?

Then we can do a final review of the code. Does that sound good?

humphd commented on src/extensions/default/brackets-show-whitespace/main.js:
```diff
@@ -0,0 +1,319 @@
+    var commandID          = "denniskehrig.ShowWhitespace.toggle";
+    var preferencesID      = "denniskehrig.ShowWhitespace";
+    var defaultPreferences = {
+        enabled: true,
```
Let's change this to `false` by default, so that we have to flip a pref in order to enable it.  Then you can load it all the time.

humphd commented:
I've updated https://github.com/mozilla/thimble.mozilla.org/issues/1763#issuecomment-287172964 to be about the UI toggle.

humphd commented:
@tonypark0403 when you add new commits and it's ready for review again, you should @ mention the person who is going to do the review so they know.  Otherwise we lose track.  I'll review this now.

humphd commented on src/extensions/default/brackets-show-whitespace/styles/whitespace-colors-css.tmpl:
```diff
@@ -0,0 +1,31 @@
+/*
```
@flukeout do you know what a `.tmpl` file is with CSS?  I've never seen this before.

humphd commented:
Tested this locally, it works great.  Nice work @tonypark0403.  Now you can do the Thimble side and add UI to enable/disable it.

humphd commented:
@tonypark0403 so here is my suggestion for you:

* 0.2 is this PR that adds the extension to Brackets, but it's default to `enabled=false`
* 0.3 is the part in Brackets that adds `bramble.enableWhiteSpace()` and `bramble.disableWhiteSpace()` which is similar to [this](https://github.com/mozilla/brackets/commit/b524178098f17ccf129268daf4664493916835cb)
* 0.4 is the Thimble UI in the menu that calls what you wrote in 0.3 and would be done in https://github.com/mozilla/thimble.mozilla.org/issues/1763
